Assets/Scripts/UI/CameraIntMsg.cs

```diff
@@ -3,41 +3,39 @@
 
 public class CameraIntMsg : MonoBehaviour {
 
-    public GameObject Camera;
+    public GameObject CameraSprite;
     public GameObject ArrowsSide;
     public GameObject ArrowsUp;
-    public Controller2 isghostMode;
+    private Controller2 ghostMode;
     private bool onlyOnce;
 
     // Use this for initialization
     void Start()
     {
         onlyOnce = false;
+        ghostMode = GetComponentInChildren<Controller2>();
```

Comment:
also, have you tried testing if you switch to another doll, the UI still follow the current doll? from what I see here is that you never update the Controller2 after start.  Try to look at cameraFollow.cs or even LevelManager.cs as those 2 already keep tracks of which game object is the player